config/configuration.go

```diff
@@ -30,9 +31,12 @@ var (
 	// DefaultUnixLogLocation is the primary location to find log files
 	DefaultUnixLogLocation = "/var/log/cloudflared"
 
+	// DefaultXDGConfigLocation is the primary user location to find a config file
+	DefaultXDGConfigLocation = xdg.ConfigHome + "/cloudflared"
+
 	// Launchd doesn't set root env variables, so there is default
 	// Windows default config dir was ~/cloudflare-warp in documentation; let's keep it compatible
-	defaultUserConfigDirs = []string{"~/.cloudflared", "~/.cloudflare-warp", "~/cloudflare-warp"}
+	defaultUserConfigDirs = []string{DefaultXDGConfigLocation, "~/.cloudflared", "~/.cloudflare-warp", "~/cloudflare-warp"}
```

Comment:
This should only be appended to dirs if the environment is xdg

Reply:
the xdg spec says that if the enviornment variable is not set the default location of `$HOME/.config` should be used.
Many linux distros/users do not set the env and expect this behavior.

Thats also the primary reason I used the lib and do not read directly from the env as every platform specifies its own standard dir. Checkout the libs readme to see the ones used.